android/src/main/java/com/reactnativevosk/VoskModule.kt

```diff
@@ -23,10 +24,8 @@ class VoskModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaMo
     val text = getHypothesisText(hypothesis)
 
     // Stop recording if data found
-    if (text != null && text.isNotEmpty() && speechService != null) {
-      speechService!!.stop()
-      speechService = null
-      // Display data
+    if (text != null && text.isNotEmpty()) {
+      cleanRecognizer();
```

Comment:
@riderodd I am facing a problem where the `onFinalResult` event is never triggered, and I believe this change is the reason since everything is cleaned up on the first result. Does it make sense?